modules/vector-search/pages/vector-search-index-architecture.adoc

```diff
@@ -0,0 +1,151 @@
+All changes from Database Change Protocol (DCP) and the Data Service are introduced to a Search index in batches, which are further managed by segments. 
+
+[#sync]
+== Synchronization with Database Change Protocol (DCP) and the Data Service
+
+The Search Service uses batches to process data that comes in from xref:server:learn:clusters-and-availability/intra-cluster-replication.adoc#database-change-protocol[DCP] and the xref:cloud:clusters:databases.adoc#couchbase-services[Data Service].
+DCP and Data Service changes are introduced gradually, based on available memory on Search Service nodes, until reindexing operations for an index are complete.
+
+The Search Service can merge batches into a single batch before they're sent to the disk write queue, to reduce the resources required for batch processing. 
+
+The Search Service maintains index snapshots on each Search index partition.
+These snapshots contain a representation of document mutations on either a write queue, or in storage.
+
+If the Search Service loses connection to the Data Service, the Search Service compares its rollback sequence numbers in its snapshots with the Data Service when the connection is reestablished.
+If the index snapshots on the Search Service are too far ahead, the Search Service performs a full rollback to get back in sync with the Data Service. 
+
+[#segments]
+== Search Index Segments
+
+Search and Vector Search indexes in Couchbase Server are built with segments. 
+
+All Search indexes contain a root segment, which includes all data for the Search index but excludes any segments that might be stale.
+Stale segments are eventually removed by the Search Service's persister or merger routines.
+
+The persister reads in-memory segments from the disk write queue and flushes them to disk, completing batch operations as part of <<sync,>>.
+The merger works with the persister to consolidate flushed files and flush the consolidated results back through the persister - while purging the smaller, older files.
+
+The persister and merger interact to continuously flush and merge new in-memory segments to disk, and remove stale segments.
+
+Segments are marked as stale when they're replaced by a new merged segment created by the merger. 
+Stale segments are deleted when they're no longer used by any new queries. 
+
+As smaller segments are merged together through the merger routine, the Search Service automatically runs any needed retraining for Vector Search indexes.
+The segments for a Vector Search index can contain different index types and use a separate indexing pipeline, choosing the appropriate indexing algorithm based on the size of your available documents.
+
+== Vector Search and FAISS
+
+Vector Search specifically uses https://faiss.ai/index.html[FAISS^] indexes.
+Any vectors inside your documents are indexed using FAISS, to create a new query vector that can be searched for similar vectors inside your Vector Search index.
```

Comment:
Worth mentioning here the faiss indexes are embedded within segments alongside the text indexes and other data structures.